src/cs_transmog.cpp

```diff
@@ -103,31 +108,52 @@ class transmog_commandscript : public CommandScript
         }
 
         uint32 itemId = itemTemplate->ItemId;
-        uint32 accountId = target->GetSession()->GetAccountId();
+        uint32 accountId;
+        auto guid = player->GetGUID().GetCounter();
+        if (QueryResult result = CharacterDatabase.Query("SELECT `account` FROM `characters` WHERE `guid` = {}", guid))
+        {
+            Field* fields = result->Fetch();
+            accountId = fields[0].Get<uint32>();
+        }
+        else
+        {
+            return false;
+        }
+
         std::stringstream tempStream;
         tempStream << std::hex << ItemQualityColors[itemTemplate->Quality];
         std::string itemQuality = tempStream.str();
         std::string itemName = itemTemplate->Name1;
-        std::string nameLink = handler->playerLink(target->GetName());
+
+        bool isNotConsoleAndIsPlayerOnline = handler->GetSession() && target->GetSession();
```

Comment:
This causes a crash when using the command from an in-game (GM) Player onto an offline Target-Player.
In console it works because `handler->GetSession()` returns false and `target->GetSession()` isn't being executed.
But with a null-target, `target->GetSession()` causes a Segmentation fault.

I'm testing some solutions and will edit my review once I got the optimal one.

Reply:
The big problem is that `player` is a `PlayerIdentifier` and not actually a `Player` object. So a few methods cause issues here.

A few issues can be fixed with workarounds and for others we could just remove the problem by adding a check whether the player is online or not first.
The 2 issues I cannot find an obvious workaround for are the following 2 **required** checks. Since they are important, we shouldn't just remove them.

https://github.com/azerothcore/mod-transmog/blob/1a569e34bbc9ec084da5db14ffb1302249c8b2ea/src/cs_transmog.cpp#L96

https://github.com/azerothcore/mod-transmog/blob/1a569e34bbc9ec084da5db14ffb1302249c8b2ea/src/cs_transmog.cpp#L136

Might have to go back to the old system of getting the player over target or name or extend the core system.